tsdb/engine/tsm1/reader.go

```diff
@@ -127,13 +127,10 @@ func (b *BlockIterator) Next() bool {
 		}
 	}
 
-	for b.n-b.i > 0 {
+	if b.n-b.i > 0 {
```

Comment:
It was the wrong fix I added in #6501.  I'm reverting it here because the PR has the right fix down in the the tombstones.